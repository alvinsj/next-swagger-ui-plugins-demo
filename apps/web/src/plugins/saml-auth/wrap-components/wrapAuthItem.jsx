import React from "react"

const wrapAuthItem = (Ori, system) => function WrappedSamlAuthItem({ Ori, ...props }) {
  const {
    schema, getComponent, errSelectors, authorized, onAuthChange, name
  } = props

  const SamlAuth = getComponent("SamlAuth")
  const type = schema.get("type")
  const scheme = schema.get("scheme")
  const saml = schema.get("saml")

  if(type === "http" && scheme == 'bearer' && !!saml) {
    return <SamlAuth key={ name }
              schema={ schema }
              name={ name }
              authSelectors={system.getSystem().authSelectors}
              errSelectors={ errSelectors }
              authorized={ authorized }
              getComponent={ getComponent }
              onChange={ onAuthChange }
              getSystem={system.getSystem} />
  }
  else if(!Ori) return <div />
  else {
    return <Ori {...props} />
  }
}

export default wrapAuthItem
