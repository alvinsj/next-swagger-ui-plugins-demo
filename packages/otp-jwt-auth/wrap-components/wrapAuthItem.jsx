import React from "react"

const wrapAuthItem = (Ori, system) => function WrappedOtpJwtAuthItem({ ...props }) {
  const {
    schema, getComponent, errSelectors, authorized, onAuthChange, name
  } = props

  const OtpJwtAuth = getComponent("OtpJwtAuth")
  const type = schema.get("type")
  const scheme = schema.get("scheme")
  const otp = schema.get("otp")
  
  if(type === "http" && scheme === 'bearer' && !!otp) {
    return <OtpJwtAuth key={ name }
              schema={ schema }
              name={ name }
              authSelectors={system.getSystem().authSelectors}
              errSelectors={ errSelectors }
              authorized={ authorized }
              getComponent={ getComponent }
              onChange={ onAuthChange }
              getSystem={system.getSystem} />
  }
}

export default wrapAuthItem
