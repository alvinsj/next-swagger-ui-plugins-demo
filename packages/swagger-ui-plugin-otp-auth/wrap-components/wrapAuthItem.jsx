import React from "react"

const wrapAuthItem = (Ori, system) => function WrappedOtpAuthItem({ ...props }) {
  const {
    schema, getComponent, errSelectors, authorized, onAuthChange, name
  } = props

  const OtpAuth = getComponent("OtpAuth")
  const type = schema.get("type")
  const scheme = schema.get("scheme")
  const otp = schema.get("otp")
  
  if(type === "http" && scheme === 'bearer' && !!otp) {
    return <OtpAuth key={ name }
              schema={ schema }
              name={ name }
              authSelectors={system.getSystem().authSelectors}
              errSelectors={ errSelectors }
              authorized={ authorized }
              getComponent={ getComponent }
              onChange={ onAuthChange }
              getSystem={system.getSystem} />
  }
}

export default wrapAuthItem
