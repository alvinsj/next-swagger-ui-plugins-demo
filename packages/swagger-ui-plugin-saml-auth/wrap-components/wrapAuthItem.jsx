import React from "react"

const wrapAuthItem = (Ori, system) => function WrappedSamlAuthItem({ ...props }) {
  const {
    schema, getComponent, errSelectors, authorized, onAuthChange, name
  } = props

  const SamlAuth = getComponent("SamlAuth")
  const type = schema.get("type")
  const scheme = schema.get("scheme")
  const saml = schema.get("saml")

  if(type === "http" && scheme == 'bearer' && !!saml) {
    return <SamlAuth key={ name }
              schema={ schema }
              name={ name }
              authSelectors={system.getSystem().authSelectors}
              errSelectors={ errSelectors }
              authorized={ authorized }
              getComponent={ getComponent }
              onChange={ onAuthChange }
              getSystem={system.getSystem} />
  }

}

export default wrapAuthItem
