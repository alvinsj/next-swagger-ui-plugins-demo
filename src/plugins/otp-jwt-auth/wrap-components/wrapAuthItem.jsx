import React from "react"

const wrapAuthItem = (Ori, system) => function WrappedOtpJwtAuthItem({ Ori, ...props }) {
  const {
    schema, getComponent, errSelectors, authorized, onAuthChange, name
  } = props

  const OtpJwtAuth = getComponent("OtpJwtAuth")
  const type = schema.get("type")
  const scheme = schema.get("scheme")
  const otp = schema.get("otp")
  
  if(type === "http" && scheme === 'bearer' && !!otp) {
    return <OtpJwtAuth key={ name }
              schema={ schema }
              name={ name }
              authSelectors={system.getSystem().authSelectors}
              errSelectors={ errSelectors }
              authorized={ authorized }
              getComponent={ getComponent }
              onChange={ onAuthChange }
              getSystem={system.getSystem} />
  }
  else if(!Ori) return <div />
  else {
    return <Ori {...props} />
  }
}

export default wrapAuthItem
