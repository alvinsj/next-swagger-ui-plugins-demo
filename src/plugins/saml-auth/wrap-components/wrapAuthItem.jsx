import React from "react"

const wrapAuthItem = (Ori, system) => function WrappedAuthItem({ Ori, ...props }) {
  const {
    schema, getComponent, authSelectors, errSelectors, authorized, onAuthChange, name
  } = props

  const SamlAuthItem = getComponent("SamlAuthItem")
  const type = schema.get("type")
  const scheme = schema.get("scheme")

  if(type === "http" && scheme == 'bearer') {
    return <SamlAuthItem key={ name }
              schema={ schema }
              name={ name }
              authSelectors={authSelectors}
              errSelectors={ errSelectors }
              authorized={ authorized }
              getComponent={ getComponent }
              onChange={ onAuthChange }
              getSystem={system.getSystem} />
  } else {
    return <Ori {...props} />
  }
}

export default wrapAuthItem
