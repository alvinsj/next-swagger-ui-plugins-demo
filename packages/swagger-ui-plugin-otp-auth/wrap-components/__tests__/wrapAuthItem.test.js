import React from "react"
import wrapAuthItem from "../wrapAuthItem"
import renderer from "react-test-renderer"
import { Map } from "immutable"

describe("wrapAuthItem", () => {
  const system = {
    getSystem: () => system,
    otpJwtAuthSelectors: {},
    authSelectors: {},
    errSelectors: {},
    getComponent: (name) =>
      function MockComponent({ children, ...props }) {
        return (
          <mock-component id={name} data-props={props}>
            {children}
          </mock-component>
        )
      },
  }
  const Ori = () => <mock-element id={"Ori"} />
  const WrappedComponent = wrapAuthItem(Ori, system)

  it("returns null component for others", () => {
    expect(WrappedComponent).toBeInstanceOf(Function)

    // match snapshot
    const tree = renderer
      .create(
        <WrappedComponent schema={new Map()} name="OtpAuth" {...system} />
      )
      .toJSON()
    expect(tree).toMatchInlineSnapshot(`null`)
  })

  it("returns an OtpAuth component", () => {
    expect(WrappedComponent).toBeInstanceOf(Function)

    // match snapshot
    const tree = renderer
      .create(
        <WrappedComponent
          schema={new Map({ type: "http", scheme: "bearer", otp: true })}
          name="OtpAuth"
          {...system}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
