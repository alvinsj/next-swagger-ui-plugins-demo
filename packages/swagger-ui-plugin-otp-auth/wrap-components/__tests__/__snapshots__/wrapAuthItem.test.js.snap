// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapAuthItem returns an OtpAuth component 1`] = `
<mock-component
  data-props={
    {
      "authSelectors": {},
      "authorized": undefined,
      "errSelectors": {},
      "getComponent": [Function],
      "getSystem": [Function],
      "name": "OtpAuth",
      "onChange": undefined,
      "schema": Immutable.Map {
        "type": "http",
        "scheme": "bearer",
        "otp": true,
      },
    }
  }
  id="OtpAuth"
/>
`;
